tests_integration/test_git.py

```diff
@@ -0,0 +1,237 @@
+import pytest
+from snowflake.connector.errors import ProgrammingError
+from pathlib import Path
+import tempfile
+
+FILE_IN_REPO = "RELEASE-NOTES.md"
+
+
+@pytest.fixture
+def sf_git_repository(runner, test_database):
+    repo_name = "SNOWCLI_TESTING_REPO"
+    integration_name = "SNOW_GIT_TESTING_API_INTEGRATION"
+
+    if not _integration_exists(runner, integration_name=integration_name):
+        result = runner.invoke_with_connection(
+            [
+                "sql",
+                "-q",
+                f"""
+                CREATE API INTEGRATION {integration_name}
+                API_PROVIDER = git_https_api
+                API_ALLOWED_PREFIXES = ('https://github.com/snowflakedb/')
+                ALLOWED_AUTHENTICATION_SECRETS = ()
+                ENABLED = true
+            """,
+            ]
+        )
+        assert result.exit_code == 0
+
+    result = runner.invoke_with_connection(
+        [
+            "sql",
+            "-q",
+            f"""
+            CREATE GIT REPOSITORY {repo_name}            
+            API_INTEGRATION = {integration_name}
+            ORIGIN = 'https://github.com/snowflakedb/snowflake-cli.git'   
+            """,
+        ]
+    )
+    assert result.exit_code == 0
+    return repo_name
+
+
+@pytest.mark.integration
+def test_object_commands(runner, sf_git_repository):
+    # object list
+    result = runner.invoke_with_connection_json(["object", "list", "git-repository"])
+    assert result.exit_code == 0
+    assert sf_git_repository in _filter_key(result.json, key="name")
+
+    # describe
+    result = runner.invoke_with_connection_json(
+        ["object", "describe", "git-repository", sf_git_repository]
+    )
+    assert result.exit_code == 0
+    assert result.json[0]["name"] == sf_git_repository
+
+    # drop
+    result = runner.invoke_with_connection_json(
+        ["object", "drop", "git-repository", sf_git_repository]
+    )
+    assert result.exit_code == 0
+    assert result.json == [{"status": f"{sf_git_repository} successfully dropped."}]
+
+
+@pytest.mark.integration
+def test_fetch(runner, sf_git_repository):
+    result = runner.invoke_with_connection_json(["git", "fetch", sf_git_repository])
+    assert result.exit_code == 0
+    assert result.json == [
+        {
+            "status": f"Git Repository {sf_git_repository} is up to date. No change was fetched."
+        }
+    ]
+
+
+@pytest.mark.integration
+def test_list_branches_and_tags(runner, sf_git_repository):
+    # list branches
+    result = runner.invoke_with_connection_json(
+        ["git", "list-branches", sf_git_repository]
+    )
+    assert result.exit_code == 0
+    assert "main" in _filter_key(result.json, key="name")
```

Comment:
HAve you considered using `like` to reduce the json and do explicit comparison ?

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1223142